bot/utils/Interactions.py

```diff
@@ -0,0 +1,45 @@
+from typing import Optional
+
+from discord import Interaction, enums, ui
+
+from bot.constants import MODERATION_ROLES
+
+
+class InvokerAndModeratorView(ui.View):
+    """A view that allows the original invoker and moderators to interact with it."""
+
+    def __init__(self, invoker_id: int, timeout: Optional[float] = 180.0) -> None:
+        super().__init__(timeout=timeout)
+        self.invoker_id = invoker_id
+
+    async def interaction_check(self, interaction: Interaction) -> bool:
+        """Ensure the user clicking the button is the view invoker, or a moderator."""
+        if self.invoker_id == interaction.user.id:
+            return True
+
+        if any(role.id in MODERATION_ROLES for role in getattr(interaction.user, "roles", [])):
+            return True
+
+        await interaction.response.send_message("This is not your button to click!", ephemeral=True)
+        return False
+
+
+class DeleteMessageButton(ui.Button):
+    """
+    A button that can be added to a view to delete the message containing the view.
+
+    This button itself carries out no interaction checks, this should be done by the parent view.
```

Comment:
```suggestion
    This button itself carries out no interaction checks; that should be done by the parent view.
```

Reply:
This has been moved to bot-core with slightly different wording.
https://github.com/python-discord/bot-core/blob/main/botcore/utils/interactions.py#L65